src/ws-header/simple-steam.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * This utility class implements a simple stream to listen continuously for changes
+ */
+export class SimpleSteam {
```

Comment:
Thx for the hint :)
I had the problem that multiple parts in the authorization can change the access token multiple times. So there were two ways. Either I'll pass a callback to the constructor header or I'll go with a reactive stream approach. At the point where I implemented this the code looked a bit different and this was the most attractive way for me. Right now I probably would go with the callback approach since it is much simpler to understand for others.